test/fail_compilation/constraints_func4.d

```diff
@@ -0,0 +1,97 @@
+/*
+EXTRA_FILES: imports/constraints.d
+REQUIRED_ARGS: -verrors=context
+TEST_OUTPUT:
+---
+fail_compilation/constraints_func4.d(90): Error: template `imports.constraints.overload` cannot deduce function from argument types `!()(int)`
+    overload(0);
+            ^
+fail_compilation/imports/constraints.d(39):        Candidates are: `overload(T)(T v)`
+  with `T = int`
+  must satisfy the following constraint:
+`       N!T`
```

Comment:
```
`       N!T`
```
I see that the formatter could still do with some improvements, but this is nothing related to this change here.